Playing albums without shuffle

FriederHannenheim opened the pull request:
I've implemented `CoverSliverAppBar` to provide the button on the album page. I also implemented the toggle in the settings but the stream in settings_dao doesn't update sadly. Could you please take a look at it to find out why?
I still need to implement the function of the setting. It's just the toggle for now.

moritz-weber commented on src/lib/system/datasources/moor_database.dart:
```diff
@@ -240,7 +240,12 @@ class MoorDatabase extends _$MoorDatabase {
+            );
+            await into(keyValueEntries).insert(
+              const KeyValueEntriesCompanion(
+                  key: Value(SETTING_ALLOWED_EXTENSIONS), value: Value('false')),
```
```suggestion
                  key: Value(SETTING_PLAY_ALBUMS_IN_ORDER), value: Value('false')),
```
Small copy paste error here.

moritz-weber commented on src/lib/system/repositories/settings_repository_impl.dart:
```diff
@@ -14,6 +14,7 @@ class SettingsRepositoryImpl implements SettingsRepository {
 
   final BehaviorSubject<bool> _manageExternalStorageGrantedSubject = BehaviorSubject();
   final BehaviorSubject<String> _fileExtensionsSubject = BehaviorSubject();
+  final BehaviorSubject<bool> _playAlbumsInOrderSubject = BehaviorSubject();
```
You need to add a listener to the stream of the data source. Same as in line 10:

    _settingsDataSource.playAlbumsInOrderStream.listen(_playAlbumsInOrderSubject.add);

The rationale is that this way, we transform the normal stream into a `ValueStream` that has (IMO) some beneficial properties for the usage in the UI. This setup might be a bit convoluted, but so far, it worked well for me and I keep to this recipe.

moritz-weber commented:
Just two small errors, nothing wild. The rest looks fine to me (just skimmed through the changes briefly). One thing I noticed was that the background image in the album header is missing. You should be able to fix this with something like this as background widget: https://github.com/moritz-weber/mucke/blob/a57b16011722684e704684af3b0133dae6a1f86a/src/lib/presentation/widgets/album_sliver_appbar.dart#L288-L305

FriederHannenheim commented:
I implemented the setting and fixed some small oversights that I found. Please let me know if you like my implementation and what you would change. 

Some things I've noticed on the way:
- some lines still get reformatted even though I have set my line width to 100
- I've now added the artist name back to the album page and with long names it sometimes overlaps the play button

FriederHannenheim commented on src/lib/presentation/pages/album_details_page.dart:
```diff
@@ -58,10 +66,98 @@ class _AlbumDetailsPageState extends State<AlbumDetailsPage> {
+                      child: Image(
+                        image: utils.getAlbumImage(album.albumArtPath),
+                        fit: BoxFit.cover,
+                        opacity: const AlwaysStoppedAnimation(1.0),
```
There's some function calculating this value in the old AlbumSliverAppBar. I decided to drop it since I didn't see much of a difference. If it has some use that I missed I can add it back.

moritz-weber replied:
You're right. I don't see a difference either. We can even omit this parameter altogether.

FriederHannenheim commented:
@moritz-weber forgot to ping you

moritz-weber commented:
No problem, I get notified anyway. Just didn't have time to check it yet. Wednesday probably.

moritz-weber commented:
Thank you for your work, I really like the implementation. I'll probably play around with the components in the header to figure out a way to avoid the overlapping artist name.

FriederHannenheim commented:
Thanks. It's really fun working in Flutter. Would you mind sharing your To-Do list for this project? I'd really like to contribute more.

moritz-weber commented:
Sure, no problem. If you send me an e-mail (just made my address visible in my profile), I can contact you directly. Although my to-do list is more like a loose collection of things I might want to do with the app some day.
